sections/subsections/controlflow2.tex

```diff
@@ -79,9 +88,57 @@ \subsection{if, if-else and ?:} \todo{Jeg blander meget rundt i statement og con
 
 An example of the ternary operator \texttt{?:} is shown in \myref{terlst}. 
 If the boolean statement before the question mark is true, then the first statement is executed otherwise the second is. 
-The ternary operator can also be used for an assignment. 
+The ternary operator can also be used for an assignment, only if the value getting assigned to is left of the question mark.
+As the ternary operator allows any statement in its middle body, it is possible to chain them together, just like a if-else chain. 
+
+\subsection{while}
+An example of the while loop in GAMBLE is shown in \myref{whilelst}. 
+The syntax is identical to the if statement, which also makes it look like a function in certain ways. 
+It's also identical to the syntax found in many other programming languages and is therefore similar to many programmers. \todo{Skal sådan et argument være med?}
+
+\noindent\begin{minipage}{.40\textwidth}
+\begin{lstlisting}[caption=A while loop in GAMBLE.,frame=tlrb, label=whilelst, numbers=none]{Name}
+while ( condition )
+{
+    statement();
+}
+\end{lstlisting}
+\end{minipage}\hfill
+\begin{minipage}{.50\textwidth}
+\begin{lstlisting}[caption=A for loop in GAMBLE.,frame=tlrb, label=forlst, numbers=none, language=C]{Name}
+for ( initialize; condition; update )
+{
+    statement();
+}
+\end{lstlisting}
+\end{minipage}
+
+\subsection{for}
+An example of the for loop in GAMBLE is shown in \myref{forlst}.
+An alternative to this syntax is one which allows an iteration over a collection such as a foreach or one that iterates over a range.
+Examples of how such a syntax should be is shown in \myref{forextlst}.
+
+\begin{lstlisting}[caption=Alternative syntaxes for the for loop.,frame=tlrb, label=forextlst, numbers=none, language=C]{Name}
+// iterates over every int in the range.
+for ( int i in range 0:10 )
+{
+    statement(i);
+}
+// iterates over every other int in the range.
+for ( int i in range 0:10:2 )
+{
+    statement(i);
+}
+// iterates over each element as a reference to the element in the collection
+foreach ( item i in collection )
+{
+    statement(i);
+}
+// iterates over each index in a collection
+foreach ( index i in collection )
+{
+    statement(collection[i]);
+}
+\end{lstlisting}
 
-\subsection{while} 
-\todo{Er ikke sikker på om den metode anvendt til if er fik og/eller den vi vil bruge. Venter med at skrive dette.}
-\subsection{for} 
-\todo{Er ikke sikker på om den metode anvendt til if er fik og/eller den vi vil bruge. Venter med at skrive dette.}
\ No newline at end of file
+These each have the strength of expressing clearly what they do, however they are easily expressed in the form expressed in \myref{forlst}.

```

Comment:
easily er subjektivt, det lyder også underligt med 2*expressed så hurtigt efter hinanden.
"they can be expressed  in the form shown in myref and as such they are not deemed necessary."